exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -258,6 +251,13 @@ protected boolean prefetchFirstBatchFromBothSides() {
       return false;
```

Comment:
`checkForEarlyFinish` checks for NONE from both side. What if only one side receives NONE? In that case also `BatchMemoryManager.update()` should not be called

Reply:
The case of only one side receiving NONE is legit -- for those RIGHT/LEFT/FULL OUTER joins, where the other side is still returned.
As for the update() on a NONE -- probably any setting of NONE also initializes the container's record count to zero.

Reply:
AFAIK NONE is not supposed to have a valid incoming container associated with it

Reply:
@Ben-Zvi I agree the case where only one side receives NONE is legit. I think the issue @sohami was referring to is that **batchMemoryManager.update** will be called for both sides even when only one side has NONE. Looking at the javadoc for IterOutcome.NONE there is no guarantee that we will have a container with a value record count set on it. Also looking at other operators such as project, they do not seem to set the record count in the case where they process an empty table, so we cannot assume that a container will have a record count of 0 when receiving NONE.

Could you add a check to call update and debug statements for the left and right side only if their IterOutcome is not NONE?